cmd/apiserver/app/server/run_server.go

```diff
@@ -42,7 +42,8 @@ func RunServer(opts *ServiceCatalogServerOptions) error {
 func runTPRServer(opts *ServiceCatalogServerOptions) error {
 	tprOpts := opts.TPROptions
 	glog.Infoln("Installing TPR types to the cluster")
-	if err := tpr.InstallTypes(tprOpts.clIface); err != nil {
+	tprInstaller := tpr.NewInstaller(tprOpts.clIface.Extensions().ThirdPartyResources())
```

Comment:
my suggestion was to go back to an exported function that takes a third party resources interface. testers can easily pass a mock TPR interface to that function. less production and test code, and easier to follow.